### DISCUSSION TITLE
One container per task or for all tasks

### DISCUSSION THREAD

**Discussion by ncouronne** [flyteorg/flyte#1241] [Q&A] [answered]
Hi,

I'm currently evaluating flyte for managing our workflows and tasks which are currently based on docker container.
flyte seems to be very promising matching many of our needs.
We have some workflows where each task need to be executed in a single container. At the opposite, other workflows, due to the huge amount of data shared between tasks, need to be executed in a single container gathered all the tasks.

It's not clear for me if flyte (service) can handle a workflow as a single container. 
The default behaviour in flyte is one task = one container, but in some cases, i would like to execute all tasks or maybe some of them in a single container, how can i configure the workflow to do so ?

Thanks in advance

**Comment by kumare3** [comment, marked as answer]
@ncouronne IIUC, you would like in some cases that the entire workflow simply runs in one container. Currently, Flyte does not support automatically fusing multiple tasks into a single container. But you can achieve this using Subworkflows and Local workflow execution. At the moment, from Flyte's point of view this will look like a single task - wherein you can execute the subworkflow in Local Mode and then return. So in case of retries the entire "task + Subworkflow" will be retried.

To reduce the penalty of a re-run we are working on [Intratask checkpointing](https://github.com/flyteorg/flyte/issues/634) and eventually we will be supporting [Automatic Task fusion or Coalesce](https://github.com/flyteorg/flyte/issues/649)

We would love to work with you as a community and try and deliver these features. As a community we believe in building features for  users and usecases, that way we can get iterative and quick feedback.

Please join our slack channel and we would love to work with you

**Reply by kumare3** [reply]
Also to clarify to execute a workflow in local mode you just have to invoke it - within a task.

**Reply by ncouronne** [reply]
@kumare3 when you speak about local, you mean with the sandbox?
The two feature you mentioned seems promising.
A sigle container for multitask is not a crucial requirement for us for the moment.

i'd like to join the slack channel, but it seems that someone shall invit me :(

**Reply by kumare3** [reply]
Ohh that's not the case, which link did you use? In any case you can use [this link](https://docs.google.com/forms/d/e/1FAIpQLScWPJZncyL-bBCWbuQ9HWuMzA8-r2RJc__CSWJoAYUQADN-BQ/viewform)

**Reply by kumare3** [reply]
Also by local I mean, flytekit execution. Join the slack and we can try things
